packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -1585,4 +1585,116 @@ describe('ReactDOMInput', () => {
       }
     });
   });
+
+  describe('When given a Symbol value', function() {
+    it('does not allow initial assignment of Symbols to value', function() {
+      spyOnDev(console, 'error');
+      var container = document.createElement('div');
+      ReactDOM.render(<input value={Symbol('foobar')} readOnly />, container);
+      var node = container.firstChild;
+
+      expect(node.value).toBe('');
+      expect(node.getAttribute('value')).toBe('');
+
+      if (__DEV__) {
+        expect(console.error.calls.count()).toBe(1);
+        expect(console.error.calls.argsFor(0)[0]).toContain(
+          'Invalid value for prop `value`',
+        );
+      }
+    });
+
+    it('does not update the value to a Symbol', function() {
+      spyOnDev(console, 'error');
+      var container = document.createElement('div');
+      ReactDOM.render(<input value={''} readOnly />, container);
+      ReactDOM.render(<input value={Symbol('foobar')} readOnly />, container);
+      var node = container.firstChild;
+
+      expect(node.value).toBe('');
+      expect(node.getAttribute('value')).toBe('');
+
+      if (__DEV__) {
+        expect(console.error.calls.count()).toBe(1);
+        expect(console.error.calls.argsFor(0)[0]).toContain(
+          'Invalid value for prop `value`',
+        );
+      }
+    });
+
+    it('does not allow initial assignment of Symbols to defaultValue', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(<input defaultValue={Symbol('foobar')} />, container);
+      var node = container.firstChild;
+
+      expect(node.value).toBe('');
+      expect(node.getAttribute('value')).toBe('');
+    });
+
+    it('does not update defaultValue to a Symbol', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(<input defaultValue={''} />, container);
+      ReactDOM.render(<input defaultValue={Symbol('foobar')} />, container);
+      var node = container.firstChild;
+
+      expect(node.value).toBe('');
+      expect(node.getAttribute('value')).toBe('');
+    });
+  });
```

Comment:
@gaearon assigning defaultValue does not raise a warning. This is because `defaultValue` is considered a reserved prop:
https://github.com/facebook/react/blob/master/packages/react-dom/src/shared/DOMProperty.js#L12-L21

Removing it from the reserved props list addresses the issue, but then a _ton_ of server integration tests fail. The current 16.2.0 behavior also does not raise a warning:

<img width="1401" alt="screen shot 2017-12-01 at 12 50 20 pm" src="https://user-images.githubusercontent.com/590904/33495913-3b8bfc90-d696-11e7-9fdf-1eb9877d769f.png">

Do we want to address the lack of a warning for `defaultValue` in this PR?

Reply:
No, it's okay.